packages/grafana-ui/src/components/uPlot/config/gradientFills.ts

```diff
@@ -10,28 +10,61 @@ import tinycolor from 'tinycolor2';
 import uPlot from 'uplot';
 import { getCanvasContext } from '../../../utils/measureText';
 
+function makeDirectionalGradient(direction: GradientDirection, bbox: uPlot.BBox, ctx: CanvasRenderingContext2D) {
+  let x0 = 0,
+    y0 = 0,
+    x1 = 0,
+    y1 = 0;
+
+  if (direction === GradientDirection.Down) {
+    y0 = bbox.top;
+    y1 = bbox.top + bbox.height;
+  } else if (direction === GradientDirection.Left) {
+    x0 = bbox.left + bbox.width;
+    x1 = bbox.left;
+  } else if (direction === GradientDirection.Up) {
+    y0 = bbox.top + bbox.height;
+    y1 = bbox.top;
+  } else if (direction === GradientDirection.Right) {
+    x0 = bbox.left;
+    x1 = bbox.left + bbox.width;
+  }
+
+  return ctx.createLinearGradient(x0, y0, x1, y1);
+}
+
 export function getOpacityGradientFn(
   color: string,
   opacity: number
 ): (self: uPlot, seriesIdx: number) => CanvasGradient {
   return (plot: uPlot, seriesIdx: number) => {
     const ctx = getCanvasContext();
-    const gradient = ctx.createLinearGradient(0, plot.bbox.top, 0, plot.bbox.top + plot.bbox.height);
+    const gradient = makeDirectionalGradient(
+      plot.scales.x!.ori === 0 ? GradientDirection.Down : GradientDirection.Left,
+      plot.bbox,
+      ctx
+    );
 
     gradient.addColorStop(0, colorManipulator.alpha(color, opacity));
     gradient.addColorStop(1, colorManipulator.alpha(color, 0));
 
     return gradient;
   };
 }
+
 export function getHueGradientFn(
   color: string,
   opacity: number,
   theme: GrafanaTheme2
 ): (self: uPlot, seriesIdx: number) => CanvasGradient {
   return (plot: uPlot, seriesIdx: number) => {
     const ctx = getCanvasContext();
-    const gradient = ctx.createLinearGradient(0, plot.bbox.top, 0, plot.bbox.top + plot.bbox.height);
+    const gradient = makeDirectionalGradient(
+      plot.scales.x!.ori === 0 ? GradientDirection.Down : GradientDirection.Left,
```

Comment:
Can you use `ScaleOrientation` enum please?

Reply:
done